assembly/buffer/index.ts

```diff
@@ -11,6 +12,42 @@ export class Buffer extends Uint8Array {
     return new Buffer(size);
   }
 
+  public static concat<T>(items: T, length: i32): Buffer {
+    if (!isArray<T>()) {
+      ERROR("Buffer.concat<T> must accept an Array where T extends Array<Uint8Array | Buffer>");
+    }
+    assert(unchecked(items[0]) instanceof Uint8Array); // Can this be a static check?
```

Comment:
While this probably just works, potentially reading out of bounds makes me uncomfortable a bit. What this probably does here is

1. if `T extends Uint8Array`, emit a static check that also `drop`s a potentially OOB `items[0]`
2 if `T` does not extend `Uint8Array`, emit a failing dynamic check

One way to alleviate this a bit might be to only do the check if `items.length > 0`, with a more general alternative being to limit to `Buffer` as noted in the previous comment. Not 100% sure what do to.

Reply:
If the dataStart property is too far out of bounds, this will result in panic too. Will investigate removing this along with the generic parameter.

Reply:
This should be all set with the latest commit.